optaplanner-quarkus-integration/optaplanner-quarkus/runtime/src/main/java/org/optaplanner/quarkus/nativeimage/Substitute_LambdaBeanPropertyMemberAccessor.java

```diff
@@ -0,0 +1,74 @@
+package org.optaplanner.quarkus.nativeimage;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/**
+ * LambdaBeanPropertyMemberAccessor works by creating a new class during runtime (via LambdaMetafactory) to delegate to
+ * the provided getter/setter methods. This is not supported in GraalVM, so we need to use Method reflection
+ * (i.e. {@link Method#invoke(Object, Object...)}) instead.
+ */
+@TargetClass(className = "org.optaplanner.core.impl.domain.common.accessor.LambdaBeanPropertyMemberAccessor")
+public final class Substitute_LambdaBeanPropertyMemberAccessor {
+
+    @Alias
+    Method getterMethod;
+
+    @Alias
+    Method setterMethod;
+
+    @Substitute
+    private Function createGetterFunction(MethodHandles.Lookup lookup) {
+        return new GetterFunctionDelegator(getterMethod);
+    }
+
+    @Substitute
+    private BiConsumer createSetterFunction(MethodHandles.Lookup lookup) {
+        if (setterMethod == null) {
+            return null;
+        }
+
+        return new SetterFunctionDelegator(setterMethod);
+    }
+
+    private static final class GetterFunctionDelegator implements Function {
```

Comment:
I'd call it a `GetterDelegationFunction`, because it is a `Function`. Also, I'd make it a `Function<Object, Object>` - it won't do anything, but the raw type warnings will go away.